utils.py

```diff
@@ -379,11 +354,14 @@ def generate_datafiles(table, path, subj_text, dir=CACHE_DIR):
         .alias("Last Updated")
     )
 
-    # path always starts with a leading /, remove it
-    rel_path = path[1:]
-    parts = rel_path.split('/')
-    parts.append(avg_time)
-    csv_file = '-'.join(parts) + '.csv'
+    # Use a sanitized version of subj_text for the filename
+    safe_subj_text = sanitize_excel_sheetname(subj_text).replace(" ", "_")
```

Comment:
Your code for generating filenames based on the subject text didn't work because you have `\n` in the `subj_text`, I suggest the following:
```suggestion
    safe_subj_text = sanitize_excel_sheetname(subj_text).replace(" ", "_").replace("\n", "_")
```